### ISSUE TITLE
Annoying warning (‘-pedantic’ is not an option that controls warnings)

### ISSUE THREAD

**Issue by concatime** [ivmai/libatomic_ops#27]
Hi. When I compile this repo. from nginx, I get this minor warning:
```
checking for strnlen... In file included from atomic_ops.c:78:0:
atomic_ops/sysdeps/standard_ao_double_t.h:35:37: warning: ‘-pedantic’ is not an option that controls warnings [-Wpragmas]
 #     pragma GCC diagnostic ignored "-pedantic"
                                     ^~~~~~~~~~~
```
Maybe It could be relevant? Or not.

**Comment by ivmai** [comment]
Which compiler version?

**Comment by concatime** [comment]
Oh, sorry. I used `gcc 7.1.1` on Fedora.

**Comment by ivmai** [comment]
If you replace "-pedantic" to "-Wpedantic" in atomic_ops.c, does it remove the warning?

**Comment by concatime** [comment]
I'm sorry, but there is no `-pedantic` in `src/atomic_ops.c`.

**Comment by concatime** [comment]
Trying to compile directly from source using `gcc`, I get this:
```
libtool: compile:  gcc -DHAVE_CONFIG_H -I../src -I../src -fPIC -Wall -Wextra -Wpedantic -Wno-long-long -g -O2 -MT atomic_ops.lo -MD -MP -MF .deps/atomic_ops.Tpo -c atomic_ops.c  -fPIC -DPIC -o .libs/atomic_ops.o
In file included from atomic_ops.c:78:0:
atomic_ops/sysdeps/standard_ao_double_t.h:35:37: warning: ‘-pedantic’ is not an option that controls warnings [-Wpragmas]
 #     pragma GCC diagnostic ignored "-pedantic"
                                     ^~~~~~~~~~~
atomic_ops/sysdeps/standard_ao_double_t.h:37:22: warning: ISO C does not support ‘__int128’ types [-Wpedantic]
     typedef unsigned __int128 double_ptr_storage;
                      ^~~~~~~~
```

**Comment by ivmai** [comment]
Ah, sorry I meant src/atomic_ops/sysdeps/standard_ao_double_t.h.

**Comment by concatime** [comment]
Ok, cool, that works perfectly bro! It seems that `GCC before ~4.8 does not accept "-Wpedantic" quietly`. Maybe you can add a version checking? Moreover, could you provide an option to install only shared libraries like [here](https://github.com/jemalloc/jemalloc/blob/dev/INSTALL.md#advanced-compilation), it would be really useful for a lot of us.
Thank you, and I'll leave you the honor (or honour if you're more "British") to close the issue ;)

**Comment by ivmai** [comment]
Fixed in 7fad46e

**Comment by ivmai** [comment]
Request about shared-only libraries installation is moved to #28
